floor/test/integration/sqlparser_test/sqlparser_test.dart

```diff
@@ -0,0 +1,139 @@
+import 'dart:async';
+
+import 'package:floor/floor.dart';
+import 'package:floor_annotation/floor_annotation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:sqflite/sqflite.dart' as sqflite;
+
+import '../model/person.dart';
+
+part 'sqlparser_test.g.dart';
+
+void main() {
+  TestDatabase database;
+  DeepDao deepDao;
+  CaseDao caseDao;
+
+  setUp(() async {
+    database = await $FloorTestDatabase.inMemoryDatabaseBuilder().build();
+    deepDao = database.deepDao;
+    caseDao = database.caseDao;
+  });
+
+  tearDown(() async {
+    await database.close();
+  });
+
+  test('Stream works with wrong case', () async {
+    final person1 = Person(1, 'Baba');
+    final person2 = Person(2, 'Me');
+
+    final actual = caseDao.allPersons();
+    expect(
+        actual,
+        emitsInOrder(<List<Person>>[
+          [],
+          [],
+          [person1],
+          [person1],
+          [person1, person2],
+          [person1, person2],
+        ]));
+    await Future<void>.delayed(const Duration(milliseconds: 100));
+    await caseDao.updateName();
+
+    await Future<void>.delayed(const Duration(milliseconds: 100));
+    await caseDao.insertPerson(person1);
+
+    await Future<void>.delayed(const Duration(milliseconds: 100));
+    await caseDao.updateName();
+
+    await Future<void>.delayed(const Duration(milliseconds: 100));
+    await caseDao.insertPerson(person2);
+
+    await Future<void>.delayed(const Duration(milliseconds: 100));
```

Comment:
❔ I guess without these delays, the tests don't run?

Reply:
Yeah, they will not complete successfully and will have issues like the following:
```
  Expected: should do the following in order:
            • emit an event that []
            • emit an event that []
            • emit an event that [Person:Person{id: 1, name: Baba}]
            • emit an event that [Person:Person{id: 1, name: Baba}]
            • emit an event that [Person:Person{id: 1, name: Baba}, Person:Person{id: 2, name: Me}]
            • emit an event that [Person:Person{id: 1, name: Baba}, Person:Person{id: 2, name: Me}]
    Actual: <Instance of '_BroadcastStream<List<Person>>'>
     Which: emitted • []
                    • [Person{id: 1, name: Baba}]
              which didn't emit an event that [] because it emitted an event that longer than expected at location [0]
```
Because the first two (or three) statements were completed together before the stream got to re-query the database.